cpp/src/arrow/filesystem/s3fs.cc

```diff
@@ -337,6 +338,10 @@ Result<S3Options> S3Options::FromUri(const Uri& uri, std::string* out_path) {
   } else {
     options.ConfigureDefaultCredentials();
   }
+  auto endpoint_env = arrow::internal::GetEnvVar(kDefaultEndpointEnvVar);
+  if(endpoint_env.ok()) {
+    options.endpoint_override = endpoint_env.MoveValueUnsafe();
```

Comment:
Can we simplify this?

```suggestion
    options.endpoint_override = *endpoint_env;
```